Return probability density in continuous action spaces, support returning log probabilities

AdamGleave opened the pull request:
Closes #396 

I've also added a flag to toggle log or normal probabilities. Working with log probabilities will be more numerically stable in the Gaussian case, since the densities can get very small (especially if querying "off policy"; needed for things like AIRL/GAIL).

SAC's `action_probability` method is not really implemented (issues an SAC-specific warning). I've made it more consistent with the similar stub implementation in DDPG, which simplifies some test code. I'm not sure I understand the original reason why SAC does not return action probabilities, though. It seems they do exist, but are just tricky to compute because of the squashing.

araffin commented:
Hey,
I don't have much time this week... Will try to review next one ;)

araffin commented on stable_baselines/common/base_class.py:
```diff
@@ -609,46 +609,68 @@ def action_probability(self, observation, state=None, mask=None, actions=None):
+                mean, logstd = actions_proba
+                std = np.exp(logstd)
+
+                n_elts = np.prod(mean.shape[1:])
```
`[1:]` because first dimension is the batch size?

AdamGleave commented:
Thanks for the review. I've added a description to the changelog, and added a comment clarifying the `n_elts` computation (you're right, the first dimension is a batch.)
